chainerrl/agents/dqn.py

```diff
@@ -151,15 +177,16 @@ def __init__(self, q_function, optimizer, replay_buffer, gamma,
         assert batch_accumulator in ('mean', 'sum')
         self.logger = logger
         self.batch_states = batch_states
-        if episodic_update:
+        self.recurrent = recurrent
+        if self.recurrent:
```

Comment:
Could we add an assert to check that the model satisfies or doesn't satisfy the following assumption: 
```recurrent (bool): If set to True, `model` is assumed to implement
            `chainerrl.links.StatelessRecurrent` and is updated in a recurrent
            manner.```

Reply:
We can, but I wonder if having such an assert is better, since it would disallow duck-typing. If `recurrent=True`, an error will be raised anyway because recurrent and non-recurrent models have completely different interfaces.